lib/databaseProvider.js

```diff
@@ -8,38 +8,111 @@
  */
 
 const database = process.require('lib/database.js');
-const conf = process.require('lib/conf.js');
+
+const getWithNormalizedKey = (obj, normalizedKey) => {
+  for (const [key, value] of Object.entries(obj)) {
+    if (key.toLowerCase() === normalizedKey) {
+      return value;
+    }
+  }
+};
+
+const deduplicate = (entries) => {
+  const unique = [];
+  entries.forEach((entry) => {
+    if (!unique.some((item) => item.dn === entry.dn)) {
+      unique.push(entry);
+    }
+  });
+
+  return unique;
+};
+
+const getIntersection = (results) => {
```

Comment:
If entries array is empty then the whole function should also return an empty array because all conditions of the AND (&) haven't been met.

Here is a suggestion for the getIntersection function which returns no entry if not all conditions are met and also accepts more than 2 AND conditions.

```suggestion
const getIntersection = (conditionsResults) => {
  let intersectionEntries = conditionsResults.shift();

  for (const conditionResults of conditionsResults) {
    const matchingEntries = []; 

    for (const entry of conditionResults) {
      if (intersectionEntries.findIndex((item) => item.dn === entry.dn) !== -1) {
        matchingEntries.push(entry);
      }   
    };  

    intersectionEntries = matchingEntries;
  };  

  return intersectionEntries;
};
```

Reply:
Whoops, you are correct, that's what I get for staying late to write code.

Reply:
No problem!